driver/normalizer/normalizer.go

```diff
@@ -64,6 +64,25 @@ func (op opJoinNamesArray) Kinds() nodes.Kind {
 }
 
 func (op opJoinNamesArray) Check(st *State, n nodes.Node) (bool, error) {
+	// Reverse op: split string into Identifiers, load the QualifiedIdentifer and check it.
+	s, ok := n.(nodes.String)
+	if !ok {
+		return false, ErrExpectedValue.New(n)
+	}
+
+	tokens := strings.Split(string(s), "::")
+	var qual = uast.QualifiedIdentifier{}
+
+	for _, t := range tokens {
+		id := uast.Identifier{Name: t}
+		qual.Names = append(qual.Names, id)
+	}
+
+	n, err := uast.ToNode(qual.Names)
```

Comment:
Hmm, interesting. You don't need `qual` here, actually. You can declare `var names nodes.Array` and `ToNode(id)` for each element, or maybe `[]Identifier` or `[]Any` will work as well.

Reply:
Changed to `[]Identifier`.